lib/prototypes.h

```diff
@@ -448,6 +448,7 @@ extern bool valid (const char *, const struct passwd *);
 extern /*@maynotreturn@*/ /*@only@*//*@out@*//*@notnull@*/char *xmalloc (size_t size)
   /*@ensures MaxSet(result) == (size - 1); @*/;
 extern /*@maynotreturn@*/ /*@only@*//*@notnull@*/char *xstrdup (const char *);
+extern /*@null@*/void xfree(void *);
```

Comment:
Should the @null@ be by the argument?

Reply:
I fix it to ‘notnull’

Reply:
But the function does not return a value.  Instead of

```
+/*@notnull@*/void xfree(void *ap)
```

doesn't it need to be
```
+void xfree(/*@null@*/void *ap)
```